Feature/members

smcommits opened the pull request:
# Description

In this project, we have built an application for sharing posts. We have implemented a user authentication system using Devise gem.  

## Features
The app has the following features
- Posts can be created by users
- User can sign-up and login

Authentication Measures
- Users should be logged in to create a new post
- Users should be logged in to view the author's name
- 
- Validation checks for the following:
    - email is present
    - password is present

## RESTful routes
The app contains the following RESTful routes for the resources:
Posts
  - `new`
  - `create`

DeliceLydia committed:
Setup devise

- generated user model
- generated post model
- created routes

smcommits committed:
Add Post controller, Add templates for new, create and index action in
Post controller

- Added a post controller
- Added index action and created a boiler html template
- Added new action and created a template with a simple_form to create a
  new tweet
- Added a create action to create the post for current_user

DeliceLydia committed:
Add list of the posts to the index page

- Add signin, signup and signout links to the index page
- Add author name to the post when a user signed in

smcommits committed:
Update Registration, Add Styles

Registration

- Added username field in  sign up form
- Added sanitization for Sign Up forms attributes in Registration
  Controller Class
- Edited Routes to point devise to registration_controller

Style

 - Add SCSS for navbar(it is ugly though)
 - Add SCSS for posts

smcommits committed:
Update Style

- Tweaked box-shadow for all post section
- Styled the links
- Fixed page layout

DeliceLydia committed:
Style signup FORM

- Styled input field
- Styled form container
- Included login link on the signup form
- Styled call to action

smcommits committed:
Add Style

Login Form
- Added style for input fields
- Added style for button

Post Form
- Added style for input field
- Added style for body

Logic
- Added condition for displaying Signup and login button only when user
  is not signed in

DeliceLydia committed:
Fix remember me button

- Removed inside field class remember me button in scss file

smcommits committed:
Fix outline on input field

Fixed the outline issue on input fields by adding outline:none property
for every input field in the application

Add box-shadow for input fields on focus

DeliceLydia committed:
Add README

- Added information about the project
- Added tools used to build the application
- Added authors
- Added acknowledgments

smcommits committed:
Update README

- Add screenshot for the repo
- Add linked in and github profile in the author's section

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Anas Siddiqui & Delice Ingabire** 
Awesome job with your first submission for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
Just one last detail to improve before you can go ahead:

- [ ] username shouldn't be an empty string, same for post title and a post body
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

DeliceLydia committed:
Work on  validations

- Add validations for username to be unique and present
- Add validations for title and body to the post
to be present

smcommits committed:
Fix linter error for indendation

ebeagusamuel reviewed (approved):
# Project Approved :100: 
### Congratulations on getting this project approved.
##### You can now merge this branch and move to the next project.

slack(@ebeagusamuel)
